### PR TITLE
feat: webhook payloads templates on honeycombio_webhook_recipient

### PR DESCRIPTION
**Pull request by brookesargent**
## Short description of the changes

Expands the honeycombio_webhook_recipient resource to be aware of custom webhook templates. Follow-up PRs will include template variables and custom headers.

### CODE REVIEW

File: internal/provider/webhook_recipient_resource_test.go [hunk 1]
```diff
@@ -57,6 +60,98 @@ resource "honeycombio_webhook_recipient" "test" {
...
+	  type   = "trigger"
+      body = "body"
+    }
+
+	template {
+	  type   = "trigger"
+      body = "another body"
+    }
+}`, name, url),
+					ExpectError: regexp.MustCompile("Conflicting configuration arguments"),
```
Comment (brookesargent) [inline internal/provider/webhook_recipient_resource_test.go:150]: The burn alert tests seem to get the "detail" from the error and can assert against it, but I only seem to be getting the "summary" and I'm not sure why

File: client/recipient_test.go [hunk 2]
```diff
@@ -80,6 +80,57 @@ func TestRecipientsEmail(t *testing.T) {
...
 
+func TestRecipientsCustomWebhook(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	c := newTestClient(t)
+
+	body := "webhook body"
+
+	testCases := []struct {
```
Comment (brookesargent) [inline client/recipient_test.go:91]: we just have the one case for now but I was going to leave this structure we use lots of other places for ease of case additions as we go unless someone is opposed

File: internal/provider/webhook_recipient_resource.go [hunk 3]
```diff
@@ -247,3 +290,139 @@ func (r *webhookRecipientResource) Delete(ctx context.Context, req resource.Dele
...
+		NestedObject: schema.NestedBlockObject{
+			Attributes: map[string]schema.Attribute{
+				"type": schema.StringAttribute{
+					Required:    true,
+					Description: "The type of the webhook template",
+					Validators: []validator.String{
+						stringvalidator.OneOf(webhookTemplateTypes...),
+					},
+				},
+				"body": schema.StringAttribute{
```
Comment (brookesargent) [inline internal/provider/webhook_recipient_resource.go:306]: probably should do a length validator on this but I didn't find a validation in the manager to base it on so was going to hold off as a follow-up and make it match in both places

**Comment by samnottingham-hc** [comment]
This looks great! What would be a way, if any, that I could test this?

**Comment by brookesargent** [comment]
> This looks great! What would be a way, if any, that I could test this?

I added a test and that test is an actual integration test. TBH I was using the integration tests and debugger as an easy way to see my resource create/destroy appropriately. There's a bit in CONTRIBUTING.md about building the provider locally and using that build to play around with and I was struggling to get it to work this morning but maybe we could both learn/re-learn how to do this from @jharley sometime before we release this feature set so we have more confidence prior to release

File: client/recipient_test.go [hunk 4]
```diff
@@ -80,6 +80,57 @@ func TestRecipientsEmail(t *testing.T) {
 	})
 }
 
+func TestRecipientsCustomWebhook(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	c := newTestClient(t)
+
+	body := "webhook body"
```
[the remark this answers is not in the document]
Comment (brookesargent) [reply client/recipient_test.go:89]: yep, fair! adjusted this in the resource tests but forgot about it here

File: docs/resources/webhook_recipient.md [hunk 5]
```diff
@@ -19,6 +34,13 @@ The following arguments are supported:
 * `name` - (Required) The name of the Webhook Integration to create.
 * `secret` - (Optional) The secret to include when sending the notification to the webhook.
 * `url` - (Required) The URL of the endpoint to send the notification to.
+* `template` - (Optional) A configuration block (described below) to customize the webhook payload if desired.
+
+When configuring custom webhook payloads, use the `template` block, which accepts the following arguments:
```
[the remark this answers is not in the document]
Comment (brookesargent) [reply docs/resources/webhook_recipient.md:39]: Agreed, and I figure as we have public docs about variables and stuff we can probably link those here too.

File: docs/resources/webhook_recipient.md [hunk 6]
```diff
@@ -9,6 +9,21 @@ resource "honeycombio_webhook_recipient" "prod" {
   name   = "Production Alerts"
   secret = "a63dab148496ecbe04a1a802ca9b95b8"
   url    = "https://my.url.corp.net"
+
+  template {
```
[the remark this answers is not in the document]
Comment (brookesargent) [reply docs/resources/webhook_recipient.md:13]: Sure!

**Comment by samnottingham-hc** [comment]
> > This looks great! What would be a way, if any, that I could test this?
> 
> I added a test and that test is an actual integration test. TBH I was using the integration tests and debugger as an easy way to see my resource create/destroy appropriately. There's a bit in CONTRIBUTING.md about building the provider locally and using that build to play around with and I was struggling to get it to work this morning but maybe we could both learn/re-learn how to do this from @jharley sometime before we release this feature set so we have more confidence prior to release

Yup would love to get that lesson! I've been eager to get some work in the Terraform stuff but I know i'll be quiteeee rusty when I do 🤓

**Pull request merged**